app/controllers/api/public/v1/courses_controller.rb

```diff
@@ -2,16 +2,17 @@ module API
   module Public
     module V1
       class CoursesController < API::Public::V1::ApplicationController
+        include PagyPagination
         def index
-          render jsonapi: paginate(courses),
+          render jsonapi: paginated_records,
                  include: include_param,
                  class: API::Public::V1::SerializerService.call
         end
 
       private
 
-        def courses
-          @courses ||= CourseSearchService.call(
+        def pagy_scope
```

Comment:
I think this was better as it was before. `paginated_records` in the module needing `page_scope` out here in the controller doesn't seem as 'obvious' as just calling `paginate(courses)`

Reply:
opted with `paginate(scope)`